.github/workflows/build-starter-templates.yml

```
@@ -19,46 +20,65 @@ jobs:
           - tinasaurus
           - basic
         node-version: [18, 20, 22]
+    outputs:
+      failed: ${{ steps.report-errors.outputs.failed }}
 
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
-
       - name: Set up Node.js
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
 
       - uses: pnpm/action-setup@v4
         name: Install pnpm
-        if: ${{ matrix.package-manager == pnpm }}
-        id: pnpm-install
         with:
-          package_json_file: package.json
-          run_install: false
+          version: 9
+        if: ${{ matrix.package-manager == 'pnpm' }}
+
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v3
+        with:
+          hugo-version: 'latest'
+        if: ${{ matrix.template == 'tina-hugo-starter' }}
 
       - name: Run create-tina-app
-        run: |
-          npx create-tina-app@latest template-repo --pkg-manager ${{ matrix.package-manager }} --template ${{ matrix.template }}
-        working-directory: ./
+        run: npx create-tina-app@latest template-repo --pkg-manager ${{ matrix.package-manager }} --template ${{ matrix.template }}
 
       - name: Run build
         run: |
           cd template-repo
-          ${{ matrix.package-manager }} run build
+          ${{ matrix.package-manager }} install && ${{ matrix.package-manager }} run build-local
 
       - name: Report errors
+        continue-on-error: false
+        id: report-errors
         if: failure()
         run: |
           ISSUE_TITLE="Starter template '${{ matrix.template }}' failed to build with ${{ matrix.package-manager }}"
           RUN_URL="https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
 
+          echo "failed=true" >> "$GITHUB_OUTPUT"
```

```suggestion
          echo $ISSUE_TITLE >> $GITHUB_STEP_SUMMARY
          echo "failed=true" >> "$GITHUB_OUTPUT"
```
suggest adding step summaries as well